scipy/stats/_continuous_distns.py

```diff
@@ -2306,6 +2306,53 @@ def _entropy(self, c):
 gompertz = gompertz_gen(a=0.0, name='gompertz')
 
 
+class gamma_gompertz_gen(rv_continuous):
+    """ A Gamma/Gompertz continuous random variable
+    
+    %(before_notes)s
+    
+    Notes
+    -----
+    The probability density function for `gamma_gompertz` is::
+    
+        gamma_gompertz.pdf(x, c, beta) = c * exp(x) * (beta**c) / ((beta - 1 + exp(x))**(c + 1))
+    
+    for ``x >= 0``, ``c > 0``, ``beta > 0``.
+    
+    `gamma_gompertz` takes ``c`` and ``beta`` as shape parameters
+    
+    %(after_notes)s
+    
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Gamma/Gompertz_distribution    
+    
+    %(example)s
+    """
+        
+    def _logsf(self, x, c, beta):
+        # np.where is used for dealing with the case that x is too large
+        # so that sc.expm1(x) becomes infinity.
+        # Since x is very large in that case, log(beta + expm1(x))
+        # approximately equals to x
+        return c * (np.log(beta) - np.where(np.isfinite(sc.expm1(x)),
+                                            np.log(beta + sc.expm1(x)), x)) 
+    
+    def _sf(self, x, c, beta):
+        return np.exp(self._logsf(x, c, beta))
+    
+    def _cdf(self, x, c, beta):
+        return 1. - self._sf(x, c, beta)
```

Comment:
Given that the CDF can be expressed as `1 - exp(...)`, suggest actually evaluating the CDF, something like 
`-np.expm1(s*(np.log(beta) - np.log(beta - 1 + np.exp(x))))`
or
`-np.expm1(s*(np.log(beta) - x + np.log1p((beta - 1) * np.exp(-x))))`

Reply:
Thanks for your suggestion. Accordingly,
```
        return 1. - self._sf(x, c, beta)
```
needs to be changed to
```
        return 1. - (beta ** c) / (beta + sc.expm1(x))
```